[pt] Add translations for /pt/docs/concepts/distributions.md (#5125)
open-telemetry/opentelemetry.io: content/pt/docs/concepts/distributions.md

---
title: Distribuições
description: >-
  Uma distribuição, que não deve ser confundida com um fork, é uma versão
  customizada de um componente OpenTelemetry.
weight: 190
default_lang_commit: 2f34c456ab38b4d3502cd07bc36fa1455d4ef875
---

> Os projetos OpenTelemetry consistem em múltiplos [componentes](../components)

Reviewer: Pensando que aqui talvez faça sentido usar com gênero, algo como: "Os projetos **do** Opentelemetry", o que acha?

O mesmo vale pro campo `description` mais acima: "componente do OpenTelemetry".

Author: Faz sentido sim! Inclusive, me recordo de ter lido em algum lugar sobre utilizar gênero sempre que for mencionar o OpenTelemetry. Boa!

  fornecedor ou usuário final
- Pacotes ou bibliotecas adicionais que podem ser específicas para determinado
  fornecedor ou usuário final
- Cobertura adicional de testes, performance e segurança além do fornecido pelo
  OpenTelemetry
- Funcionalidades adicionais além do que já é fornecido pelo OpenTelemetry
- Remoção de funcionalidades fornecidas pelo OpenTelemetry

As distribuições geralmente se enquadram nas seguintes categorias:

- **"Pura":** Estas distribuições fornecem as mesmas funcionalidades da versão
  pública e são 100% compatíveis. As customizações normalmente aprimoram a
  facilidade de uso ou configurações. Estas customizações podem ser específicas
  para _backends_, fornecedores ou usuários finais.
- **"Plus":** Estas distribuições oferecem funcionalidades adicionais ao que é
  encontrado na _upstream_, através de componentes adicionais. Exemplos incluem
  bibliotecas de instrumentação ou exportadores de fornecedores que não foram
  incluídos na _upstream_ do OpenTelemetry.
- **"Minus":** Estas distribuições fornecem um subconjunto de funcionalidades da
  upstream. Exemplos podem incluir a remoção de bibliotecas de instrumentação ou

>   de receptores, processadores, exportadores ou extensões encontradas no projeto

Reviewer 2: ```suggestion
  de receivers, processors, exporters ou extensions encontradas no projeto
```
Lembro que fechamos um acordo de não traduzir componentes do projeto opentelemetry, não encontrei a thread. 
Atualizei o glossário: https://docs.google.com/document/d/1kyu6HgdsM3-iDyf3OuRmfw4SwNJy9BVui6nn5JsKb5I/edit

[Registro](/ecosystem/registry) que não foram incluídos na versão pública do
projeto OpenTelemetry.

## Contribuição ou distribuição? {#contribution-or-distribution}

Antes de continuar e aprender como criar sua própria distribuição, pergunte a si
mesmo se as suas adições a um determinado componente do OpenTelemetry seriam
benéficas para todos e, portanto, deveriam ser incluídas nas implementações de
referência:

- Os seus scripts são "fáceis para uso" e podem ser generalizados?
- Suas alterações nas configurações padrão podem ser a melhor opção para todos?
- Os seus pacotes ou bibliotecas adicionais, são realmente específicos?
- Sua cobertura de testes, performance ou segurança também funcionarão nas
  implementações de referência?
- Você verificou com a comunidade se as suas funcionalidades ou comportamentos
  adicionais podem se tornar parte do padrão?

## Criando a sua própria distribuição {#creating-your-own-distribution}

### Collector

Um guia sobre como criar a sua própria distribuição está disponível nesta
publicação do blog:
["Building your own OpenTelemetry Collector distribution"](https://medium.com/p/42337e994b63)

> Caso você esteja criando a sua própria distriuição, o

Reviewer: ```suggestion
Caso você esteja criando a sua própria distribuição, o
```

### Bibliotecas de instrumentação específicas para cada linguagem de programação {#language-specific-instrumentation-libraries}

Existem algumas linguagens de programação que exigem mecanismos de
extensibilidade específicos para a customização das bibliotecas de
instrumentação:

- [Java agent](/docs/zero-code/java/agent/extensions)

## Siga as diretrizes {#follow-the-guidelines}

Ao utilizar materiais relacionados ao projeto do OpenTelemetry para
distribuição, como logotipo e nome, certifique-se de estar alinhado com as
[Diretrizes de Marketing do OpenTelemetry para Organizações

> Contribuintes][diretrizes] .

Reviewer 2: ```suggestion
Contribuintes][diretrizes].
```

> Marketing do Open Telemetry para Organizações Contribuintes][diretrizes] .

Reviewer: ```suggestion
Marketing do OpenTelemetry para Organizações Contribuintes][diretrizes] .
```

---

Os projetos do OpenTelemetry consistem de múltiplos [componentes](../components)
que suportam múltiplos [sinais](../signals). As implementações de referência do
OpenTelemetry estão disponíveis em:

- [Bibliotecas de instrumentação específicas de cada linguagem de programação](../instrumentation)
- Um [binário para o Collector](/docs/concepts/components/#collector)

Qualquer implementação de referência pode ser customizada como uma distribuição.

## O que é uma distribuição? {#what-is-a-distribution}

Uma distribuição é uma versão personalizada de um componente do OpenTelemetry.
Uma distribuição é um encapsulamento em torno de um repositório do OpenTelemetry
com algumas personalizações. Distribuições não devem ser confundidas com
_forks_.

As personalizações em uma distribuição podem incluir:

- Scripts para facilitar ou personalizar o uso de um _backend_ ou fornecedor
  específico
- Alterações nas configurações padrão que são necessárias para um _backend_,
  fornecedor ou usuário final
- Pacotes ou bibliotecas adicionais que podem ser específicas para determinado
  fornecedor ou usuário final
- Cobertura adicional de testes, performance e segurança além do fornecido pelo

>   OpenTelemetry
> - Funcionalidades adicionais além do que já é fornecido pelo OpenTelemetry
> - Remoção de funcionalidades fornecidas pelo OpenTelemetry
> 
> As distribuições geralmente se enquadram nas seguintes categorias:
> 
> - **"Pura":** Estas distribuições fornecem as mesmas funcionalidades da versão
>   pública e são 100% compatíveis. As personalizações normalmente aprimoram a
>   facilidade de uso ou configurações. Estas personalizações podem ser
>   específicas para _backends_, fornecedores ou usuários finais.
> - **"Plus":** Estas distribuições oferecem funcionalidades adicionais ao que é
>   encontrado na _upstream_, através de componentes adicionais. Exemplos incluem
>   bibliotecas de instrumentação ou exportadores de fornecedores que não foram
>   incluídos na _upstream_ do OpenTelemetry.
> - **"Minus":** Estas distribuições fornecem um subconjunto de funcionalidades da
>   upstream. Exemplos podem incluir a remoção de bibliotecas de instrumentação ou
>   de receptores, processadores, exportadores ou extensões encontradas no projeto
>   do OpenTelemetry Collector. Estas distribuições podem ser fornecidas por
>   questão de suporte e/ou temas relacionados à segurança.
> 
> ## Quem pode criar uma distribuição? {#who-can-create-a-distribution}
> 
> Qualquer pessoa pode criar uma distribuição. Atualmente, diversos
> [fornecedores](/ecosystem/vendors/) oferecem
> [distribuições](/ecosystem/distributions). Além disso, usuários finais podem
> considerar criar uma distribuição caso queiram usar componentes do
> [Registry](/ecosystem/registry) que não foram incluídos na _upstream_ do projeto
> do OpenTelemetry.
> 
> ## Contribuição ou distribuição? {#contribution-or-distribution}
> 
> Antes de continuar e aprender como criar sua própria distribuição, pergunte a si
> mesmo se as suas adições a um determinado componente do OpenTelemetry seriam
> benéficas para todos e, portanto, deveriam ser incluídas nas implementações de
> referência:
> 
> - Os seus scripts para "facilitar o uso" podem ser generalizados?
> - Suas alterações nas configurações padrão podem ser a melhor opção para todos?
> - Os seus pacotes ou bibliotecas adicionais são realmente específicos?
> - Sua cobertura de testes, performance ou segurança também funcionarão nas
>   implementações de referência?
> - Você verificou com a comunidade se as suas funcionalidades ou comportamentos
>   adicionais podem se tornar parte do padrão?
> 
> ## Criando a sua própria distribuição {#creating-your-own-distribution}
> 
> ### Collector
> 
> Um guia sobre como criar a sua própria distribuição está disponível nesta
> publicação do blog:
> ["Building your own OpenTelemetry Collector distribution"](https://medium.com/p/42337e994b63)
> 
> Caso você esteja criando a sua própria distribuição, o
> [OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder)
> pode ser um bom ponto de partida.
> 
> ### Bibliotecas de instrumentação específicas para cada linguagem de programação {#language-specific-instrumentation-libraries}
> 
> Existem algumas linguagens de programação que exigem mecanismos de
> extensibilidade específicos para a personalização das bibliotecas de

Reviewer 3: ```suggestion
Uma distribuição é uma versão customizada de um componente do OpenTelemetry.
Uma distribuição é um encapsulamento em torno de um repositório do OpenTelemetry
com algumas customizações. Distribuições não devem ser confundidas com
_forks_.

As customizações em uma distribuição podem incluir:

- Scripts para facilitar ou customizar o uso de um _backend_ ou fornecedor
  específico
- Alterações nas configurações padrão que são necessárias para um _backend_,
  fornecedor ou usuário final
- Pacotes ou bibliotecas adicionais que podem ser específicas para determinado
  fornecedor ou usuário final
- Cobertura adicional de testes, performance e segurança além do fornecido pelo
  OpenTelemetry
- Funcionalidades adicionais além do que já é fornecido pelo OpenTelemetry
- Remoção de funcionalidades fornecidas pelo OpenTelemetry

As distribuições geralmente se enquadram nas seguintes categorias:

- **"Pura":** Estas distribuições fornecem as mesmas funcionalidades da versão
  pública e são 100% compatíveis. As customizações normalmente aprimoram a
  facilidade de uso ou configurações. Estas customizações podem ser
  específicas para _backends_, fornecedores ou usuários finais.
- **"Plus":** Estas distribuições oferecem funcionalidades adicionais ao que é
  encontrado na _upstream_, através de componentes adicionais. Exemplos incluem
  bibliotecas de instrumentação ou exportadores de fornecedores que não foram
  incluídos na _upstream_ do OpenTelemetry.
- **"Minus":** Estas distribuições fornecem um subconjunto de funcionalidades da
  upstream. Exemplos podem incluir a remoção de bibliotecas de instrumentação ou
  de receptores, processadores, exportadores ou extensões encontradas no projeto
  do OpenTelemetry Collector. Estas distribuições podem ser fornecidas por
  questão de suporte e/ou temas relacionados à segurança.

## Quem pode criar uma distribuição? {#who-can-create-a-distribution}

Qualquer pessoa pode criar uma distribuição. Atualmente, diversos
[fornecedores](/ecosystem/vendors/) oferecem
[distribuições](/ecosystem/distributions). Além disso, usuários finais podem
considerar criar uma distribuição caso queiram usar componentes do
[Registry](/ecosystem/registry) que não foram incluídos na _upstream_ do projeto
do OpenTelemetry.

## Contribuição ou distribuição? {#contribution-or-distribution}

Antes de continuar e aprender como criar sua própria distribuição, pergunte a si
mesmo se as suas adições a um determinado componente do OpenTelemetry seriam
benéficas para todos e, portanto, deveriam ser incluídas nas implementações de
referência:

- Os seus scripts para "facilitar o uso" podem ser generalizados?
- Suas alterações nas configurações padrão podem ser a melhor opção para todos?
- Os seus pacotes ou bibliotecas adicionais são realmente específicos?
- Sua cobertura de testes, performance ou segurança também funcionarão nas
  implementações de referência?
- Você verificou com a comunidade se as suas funcionalidades ou comportamentos
  adicionais podem se tornar parte do padrão?

## Criando a sua própria distribuição {#creating-your-own-distribution}

### Collector

Um guia sobre como criar a sua própria distribuição está disponível nesta
publicação do blog:
["Building your own OpenTelemetry Collector distribution"](https://medium.com/p/42337e994b63)

Caso você esteja criando a sua própria distribuição, o
[OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder)
pode ser um bom ponto de partida.

### Bibliotecas de instrumentação específicas para cada linguagem de programação {#language-specific-instrumentation-libraries}

Existem algumas linguagens de programação que exigem mecanismos de
extensibilidade específicos para a customização das bibliotecas de
```

Aplicando mais algumas mudanças de `personalização -> customização` que acabaram ficando pra trás.

---
title: Distribuições
description: >-
  Uma distribuição, que não deve ser confundida com um fork, é uma versão
  customizada de um componente do OpenTelemetry.
weight: 190
default_lang_commit: 2f34c456ab38b4d3502cd07bc36fa1455d4ef875
---

> Os projetos do OpenTelemetry consistem em múltiplos [componentes](../components)

Reviewer 3: ```suggestion
Os projetos do OpenTelemetry consistem de múltiplos [componentes](../components)
```

que suportam múltiplos [sinais](../signals). As implementações de referência do
OpenTelemetry estão disponíveis em:

> - [Bibliotecas de instrumentação específicas para linguagens de programação](../instrumentation)

Reviewer 3: ```suggestion
- [Bibliotecas de instrumentação específicas de cada linguagem de programação](../instrumentation)
```

> - Um [Collector binário](/docs/concepts/components/#collector)

Reviewer 3: ```suggestion
- Um [binário para o Collector](/docs/concepts/components/#collector)
```

"Binário" no final ficou parecendo um adjetivo (o Collector seria binário). O binário aqui nesse caso seria um programa, um binário que você baixa para rodar.

> Qualquer implementação de referência pode ser personalizada como uma

Reviewer 3: ```suggestion
Qualquer implementação de referência pode ser customizada como uma
```

Apesar de serem termos similares, eu acho que tem uma diferença fundamental entre os dois. Uma "personalização" é feita com base em uma pessoal ou publico alvo específico (por exemplo, uma camisa de futebol com o meu nome é personalizada). "Customização" é uma alteração feita com base nas preferências que de quem fez a mudança em si, o que se encaixaria melhor nesse caso. 

Se concordarem, seria bom atualizar os outros usos também.

Author: Ótima definição. Fiquei em dúvida entre qual utilizar. Dentro dessa definição, o "customização" fica muito mais próximo da mensagem que queremos passar. Bom!

distribuição.

## O que é uma distribuição? {#what-is-a-distribution}

Uma distribuição é uma versão personalizada de um componente do OpenTelemetry.
Uma distribuição é um encapsulamento em torno de um repositório do OpenTelemetry
com algumas personalizações. Distribuições não devem ser confundidas com
_forks_.

As personalizações em uma distribuição podem incluir:

- Scripts para facilitar ou personalizar o uso de um backend ou fornecedor
  específico

> - Alterações nas configurações padrão, necessárias para um backend, fornecedor

Reviewer 3: ```suggestion
- Alterações nas configurações padrão que são necessárias para um backend, fornecedor
```

> - Scripts para facilitar ou personalizar o uso de um backend ou fornecedor
>   específico
> - Alterações nas configurações padrão, necessárias para um backend, fornecedor
>   ou usuário final

Reviewer 3: ```suggestion
- Scripts para facilitar ou personalizar o uso de um _backend_ ou fornecedor
  específico
- Alterações nas configurações padrão, necessárias para um _backend_, fornecedor
  ou usuário final
```

- Pacotes ou bibliotecas adicionais que podem ser específicas para determinado
  fornecedor ou usuário final
- Cobertura adicional de testes, performance e segurança além do fornecido pelo
  OpenTelemetry
- Funcionalidades adicionais além do que já é fornecido pelo OpenTelemetry
- Remoção de funcionalidades fornecidas pelo OpenTelemetry

As distribuições geralmente se enquadram nas seguintes categorias:

- **"Pura":** Estas distribuições fornecem as mesmas funcionalidades da versão
  pública e são 100% compatíveis. As personalizações normalmente aprimoram a
  facilidade de uso ou configurações. Estas personalizações podem ser

>   específicas para backends, fornecedores ou usuários finais.

Reviewer 3: ```suggestion
  específicas para _backends_, fornecedores ou usuários finais.
```

- **"Plus":** Estas distribuições oferecem funcionalidades adicionais ao que é

>   encontrado na upstream, através de componentes adicionais. Exemplos incluem
>   bibliotecas de instrumentação ou exportadores de fornecedores que não foram
>   incluídos na upstream do OpenTelemetry.

Reviewer 3: ```suggestion
  encontrado na _upstream_, através de componentes adicionais. Exemplos incluem
  bibliotecas de instrumentação ou exportadores de fornecedores que não foram
  incluídos na _upstream_ do OpenTelemetry.
```

- **"Minus":** Estas distribuições fornecem um subconjunto de funcionalidades da
  upstream. Exemplos podem incluir a remoção de bibliotecas de instrumentação ou
  de receptores, processadores, exportadores ou extensões encontradas no projeto

>   do OpenTelemetry Collector. Estas distribuições podem ser fornecidas para
>   aumentar o suporte e/ou temas relacionados à segurança.

Reviewer 3: ```suggestion
  do OpenTelemetry Collector. Estas distribuições podem ser fornecidas por
  questão de suporte e/ou temas relacionados à segurança.
```

Não necessariamente o suporte seria aumentado. Como essas distribuições removem partes pode ser que o suporte como um todo seja diminuído.

## Quem pode criar uma distribuição? {#who-can-create-a-distribution}

Qualquer pessoa pode criar uma distribuição. Atualmente, diversos
[fornecedores](/ecosystem/vendors/) oferecem

> [distribuições](/ecosystem/distributions). Além disso, os usuários finais podem

Reviewer 3: ```suggestion
[distribuições](/ecosystem/distributions). Além disso, usuários finais podem
```

> considerar criar uma distribuição caso queiram usar componentes no

Reviewer 3: ```suggestion
considerar criar uma distribuição caso queiram usar componentes do
```

> [Registry](/ecosystem/registry) que não foram incluídos na upstream do projeto

Author: >**[Registry]**(/ecosystem/registry) que não foram incluídos na **upstream** do projeto

Deixando como histórico, [neste documento](https://docs.google.com/document/d/1kyu6HgdsM3-iDyf3OuRmfw4SwNJy9BVui6nn5JsKb5I/edit) eu havia levantado os pontos de tradução para **Registry** e **upstream** anteriormente. 

Eu havia traduzido, porém o que o @janssenlima comentou faz total sentido, então deixo em seus termos originais.

> [Registry](/ecosystem/registry) que não foram incluídos na upstream do projeto

Reviewer 3: ```suggestion
[Registry](/ecosystem/registry) que não foram incluídos na _upstream_ do projeto
```

do OpenTelemetry.

## Contribuição ou distribuição? {#contribution-or-distribution}

Antes de continuar e aprender como criar sua própria distribuição, pergunte a si
mesmo se as suas adições a um determinado componente do OpenTelemetry seriam
benéficas para todos e, portanto, deveriam ser incluídas nas implementações de
referência:

> - Os seus scripts são "fáceis para uso" e podem ser generalizados?

Reviewer 3: ```suggestion
- Os seus scripts para "facilitar o uso" podem ser generalizados?
```

"ease" é um verbo, então os scripts que facilitam o uso do OpenTelemetry ao invés dos scripts em si serem fáceis de usar.

- Suas alterações nas configurações padrão podem ser a melhor opção para todos?

> - Os seus pacotes ou bibliotecas adicionais, são realmente específicos?

Reviewer 3: ```suggestion
- Os seus pacotes ou bibliotecas adicionais são realmente específicos?
```

- Sua cobertura de testes, performance ou segurança também funcionarão nas
  implementações de referência?
- Você verificou com a comunidade se as suas funcionalidades ou comportamentos
  adicionais podem se tornar parte do padrão?

## Criando a sua própria distribuição {#creating-your-own-distribution}

### Collector

Um guia sobre como criar a sua própria distribuição está disponível nesta
publicação do blog:
["Building your own OpenTelemetry Collector distribution"](https://medium.com/p/42337e994b63)

Caso você esteja criando a sua própria distribuição, o
[OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder)
pode ser um bom ponto de partida.

> ### Bibliotecas de instrumentação específicas para linguagens de programação {#language-specific-instrumentation-libraries}

Reviewer 3: ```suggestion
### Bibliotecas de instrumentação específicas para cada linguagem de programação {#language-specific-instrumentation-libraries}
```

Existem algumas linguagens de programação que exigem mecanismos de
extensibilidade específicos para a personalização das bibliotecas de
instrumentação:

- [Java agent](/docs/zero-code/java/agent/extensions)

## Siga as diretrizes {#follow-the-guidelines}

Ao utilizar materiais relacionados ao projeto do OpenTelemetry para
distribuição, como logotipo e nome, certifique-se de estar alinhado com as
[Diretrizes de Marketing do OpenTelemetry para Organizações
Contribuintes][diretrizes] .

O projeto do OpenTelemetry não certifica distribuições neste momento. No futuro,
o OpenTelemetry poderá certificar distribuições e parceiros de maneira
semelhante ao que ocorre no projeto Kubernetes. Ao avaliar uma distribuição,
assegure-se de que seu uso não irá resultar em uma dependência de fornecedor

> (vendor lock-in).

Reviewer 3: ```suggestion
(_vendor lock-in_).
```

> > Qualquer suporte para uma distribuição deve ser fornecido diretamente pelos
> > seus autores e não por autores do projeto do OpenTelemetry.

Reviewer 3: ```suggestion
> Qualquer suporte para uma distribuição deve ser fornecido diretamente por quem
> a criou e não pelas pessoas envolvidas no projeto do OpenTelemetry.
```

Deixando a frase com gênero mais neutro.

Author: Perfeito! Muito bem pensado.
